api/envoy/extensions/filters/http/oauth2/v3/oauth.proto

```diff
@@ -161,6 +162,9 @@ message OAuth2Config {
   // will still process incoming ID Tokens as part of the HMAC if they are there. This is to ensure compatibility while switching this setting on. Future
   // sessions would not set the IdToken cookie header.
   bool disable_id_token_set_cookie = 17;
+
+  // Specifies the retry policy for requests to the OAuth server. If not specified, then no retries will be performed.
+  config.core.v3.RetryPolicy retry_policy = 18;
```

Comment:
Please move this to the below of `token_endpoint`. Thanks. And other is fine to me.

If you want to merge the API first, you can add the not implemented hide annotation.
Then we can treat this as an API only PR.

Reply:
@wbpcode Thanks for the comment. If it's acceptable, I'd like to include both the API and the implementation in this PR, as it's a small change.